overlord/snapstate/handlers.go

```diff
@@ -2129,6 +2133,11 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 	// Make sure if state commits and snapst is mutated we won't be rerun
 	finalStatus := state.DoneStatus
 
+	// XXX: restart handling?
+	if err := notifyLinkSnapFinished(snapsup); err != nil {
```

Comment:
it feels like this should fit into the preexisting notifyLinkParticipants mechanism, maybe we just need to pass in a bit more info